Extensions/Share/ShareViewController.swift

```diff
@@ -0,0 +1,303 @@
+        label.text = title
+
+        let checkmark = UILabel()
+        checkmark.text = "✓"
+        checkmark.font = UIFont.boldSystemFont(ofSize: 18)
+
+        [label, checkmark].forEach {
+            blue.addSubview($0)
+            $0.textColor = .white
+        }
+
+        label.snp.makeConstraints { make in
+            make.top.bottom.equalToSuperview()
+            make.left.equalToSuperview().inset(UX.pageInfoRowLeftInset)
+            make.right.equalTo(checkmark.snp.left)
+        }
+
+        checkmark.snp.makeConstraints { make in
+            make.top.bottom.equalToSuperview()
+            make.right.equalToSuperview().inset(UX.rowInset)
+            make.width.equalTo(20)
+        }
+    }
+
+    private func setupNavBar() {
+        navigationController?.navigationBar.isTranslucent = false
+        navigationController?.navigationBar.setValue(true, forKey: "hidesShadow") // hide separator line
+        navigationItem.titleView = UIImageView(image: UIImage(named: "Icon-Small"))
+        navigationItem.titleView?.contentMode = .scaleAspectFit
+        navigationItem.leftBarButtonItem = UIBarButtonItem(title: Strings.SendToCancelButton, style: .plain, target: self, action: #selector(finish))
+    }
+
+    private func setupStackView() {
+        stackView = UIStackView()
+        stackView.axis = .vertical
+        stackView.spacing = 4
+        view.addSubview(stackView)
+        stackView.snp.makeConstraints { make in
+            make.edges.equalToSuperview()
+        }
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        view.backgroundColor = .white
+
+        setupNavBar()
+        setupStackView()
+
+        let (currentPageInfoRow, pageTitleLabel, urlLabel) = makePageInfoRow()
+
+        let trailingSpace = UIView()
+
+        let rows = [
+            currentPageInfoRow,
+            makeSeparator(),
+            makeActionRow(label: Strings.ShareLoadInBackground, imageName: "menu-Show-Tabs", action: #selector(actionLoadInBackground), hasNavigation: false),
+            makeActionRow(label: Strings.ShareBookmarkThisPage, imageName: "AddToBookmarks", action: #selector(actionBookmarkThisPage), hasNavigation: false),
+            makeActionRow(label: Strings.ShareAddToReadingList, imageName: "AddToReadingList", action: #selector(actionAddToReadingList), hasNavigation: false),
+            makeSeparator(),
+            makeActionRow(label: Strings.ShareSendToDevice, imageName: "menu-Send-to-Device", action: #selector(actionSendToDevice), hasNavigation: true),
+            trailingSpace
+        ]
+
+        rows.forEach {
+            stackView.addArrangedSubview($0)
+        }
+
+        // Without some growable space at the bottom there are constraint errors because the UIView space doesn't subdivide equally, and none of the rows are growable
+        trailingSpace.snp.makeConstraints { make in
+            make.height.greaterThanOrEqualTo(1)
+        }
+
+        layoutSeparators()
```

Comment:
Setting up constraints for the separators and actionsRows can happen in `makeSeparator` and `makeActionRow` because their constraints dont depend on their superview